addons/knowledge/static/src/webclient/commands/knowledge_providers.js

```diff
@@ -0,0 +1,164 @@
+/** @odoo-module */
+
+
+import { _lt } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { HotkeyCommandItem } from "@web/core/commands/default_providers";
+import { splitCommandName } from "@web/core/commands/command_palette";
+
+const { Component } = owl;
+const { xml } = owl.tags;
+
+// Articles command
+class KnowledgeCommand extends Component {}
+KnowledgeCommand.template = xml`
+    <div class="o_command_left">
+        <icon t-att-class="'pr-2 fa ' + props.icon_string"/>
+        <span class="pr-2"><t t-slot="name"/></span>
+        <icon t-if="props.isFavourite" class="fa fa-star o_favorite pr-2"/>
+        <span t-if="props.parentName" t-esc="'— '" class="text-muted small pr-2" />
+        <span t-if="props.parentName" class="o_command_name text-muted small">
+            <t t-foreach="props.splitParentName" t-as="name" t-key="name_index">
+                <b t-if="name_index % 2" t-esc="name"/>
+                <t t-else="" t-esc="name"/>
+            </t>
+        </span>
+    </div>
+`;
+
+// "Not found, create one" command
+class Knowledge404Command extends Component {}
+Knowledge404Command.template = xml`
+    <div class="o_command_hotkey">
+        <span>
+            No Article found. <span class="text-primary">Create "<u t-esc="props.articleName"/>"</span>
+        </span>
+    </div>
+`;
+
+// Advanced search command
+class KnowledgeExtraCommand extends HotkeyCommandItem {}
+KnowledgeExtraCommand.template = xml`
+    <div class="o_command_hotkey">
+        <span>
+            <icon class="fa fa-arrows-alt pr-2" />
+            <t t-esc="props.name" />
+        </span>
+        <span>
+            <t t-foreach="getKeysToPress(props)" t-as="key">
+                <kbd t-esc="key" />
+                <span t-if="!key_last"> + </span>
+            </t>
+        </span>
+    </div>
+`;
+const commandSetupRegistry = registry.category("command_setup");
+commandSetupRegistry.add("?", {
+    emptyMessage: _lt("No article found."),
+    name: _lt("articles"),
+});
+
+const commandProviderRegistry = registry.category("command_provider");
+commandProviderRegistry.add("knowledge", {
+    namespace: "?",
+    async provide(newEnv, options) {
```

Comment:
newEnv => this ? (or simply env ?)

Reply:
I used the same syntax from `addons/mail/static/src/webclient/commands/mail_providers.js`, can change it to `this` if you want.